explore/src/main/scala/explore/common/ProposalQueries.scala

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.common
+
+import cats.syntax.all.*
+import clue.data.Input
+import clue.data.syntax.*
+import explore.model.Proposal
+import explore.model.ProposalType
+import lucuma.core.enums.ToOActivation
+import lucuma.schemas.ObservationDB.Types.ClassicalInput
+import lucuma.schemas.ObservationDB.Types.DemoScienceInput
+import lucuma.schemas.ObservationDB.Types.DirectorsTimeInput
+import lucuma.schemas.ObservationDB.Types.FastTurnaroundInput
+import lucuma.schemas.ObservationDB.Types.LargeProgramInput
+import lucuma.schemas.ObservationDB.Types.PoorWeatherInput
+import lucuma.schemas.ObservationDB.Types.ProposalPropertiesInput
+import lucuma.schemas.ObservationDB.Types.ProposalTypeInput
+import lucuma.schemas.ObservationDB.Types.QueueInput
+import lucuma.schemas.ObservationDB.Types.SystemVerificationInput
+
+trait ProposalQueries:
+  private def toOAUpdater(f: Input[ToOActivation] => Input[ToOActivation]) =
```

Comment:
Nit: This can be a `val` with type `Endo[Input[ToOActivation]] => Endo[ProposalTypeInput]`.

That way, you can omit the `f` altogether, just `X.modify >>> Y.modify ...`.